packages/typespec-azure-resource-manager/lib/legacy/models.tsp

```diff
@@ -0,0 +1,59 @@
+using TypeSpec.Http;
```

Comment:
For legacy, I think we are still leaving out scenarios where folks extend Resource or EntityResource.  I would prefer it if we just let them extend a common-types resource type, and added a new decorator they would need to decorate the type with that is the equivalent of `@armResourceInternal` and `@includeInapplicableMetadataInPayload`, and relax the rules around extension to include this case.